tensorflow_datasets/core/features/tensor_feature.py

```diff
@@ -119,6 +119,7 @@ def get_serialized_info(self):
       }
     return serialized_spec
 
+  @functools.lru_cache(maxsize=2048)
```

Comment:
example_data should be different each time. How is it supposed to help ?

Additionally, I don't think example_data is hashable. Does it actually improve the speed ?

Reply:
It does show improvement. Without running deep tests, I think that datasets with a lot of discrete features will benefit most.
Trivial example: one hot encodings have a lot of repeating 0s and 1s.

Reply:
See my comment above. Some datasets use unhashable types (e.g. lists for bounding boxes). It is, of course, fixable. The question is whether we want to convert them to hashable or not.